src/include/duckdb/storage/table/row_group_collection.hpp

```diff
@@ -154,6 +155,25 @@ class RowGroupCollection {
 	}
 	void SetAppendRequiresNewRowGroup();
 
+	std::vector<idx_t> prefix_sums;
+
+	void PrecomputePrefixSums() {
+		idx_t cumulative = 0;
+		prefix_sums.resize(owned_row_groups->GetSegmentCount());
+
+		for (idx_t i = 0; i < owned_row_groups->GetSegmentCount(); ++i) {
+			cumulative += owned_row_groups.get()->GetSegment(i)->GetNode().GetCommittedRowCount();
```

Comment:
We shouldn't be using `GetCommittedRowCount` here because this is not taking into account the snapshot we are scanning - we need to pass in the transaction id so that we can get the correct row counts for this snapshot.

The row counts could be obtained by calling `GetSelVector` (which returns the row count AND the exact row ids) - but ideally we would make a new method that allows us to get only the count for a given transaction.